spec/models/metric/purging_spec.rb

```diff
@@ -82,4 +82,24 @@
       end
     end
   end
+
+  context "#purge_realtime" do
```

Comment:
With how risky I see this change being, I feel like having more than 1 test here would be a good idea.  Didn't look myself, but based on seeing you are creating this context for `"#purge_realtime"` yourself, I am a bit discouraged that we didn't have tests around this in the first place 😞 .

Reply:
Possible edge cases that come to mind:

* Daylight savings time issues
* Different source times zones
* New years

Not saying that what you have here has bugs, but extra tests couldn't hurt with confidence.

Reply:
Along these same lines, would it make sense to write these tests in a separate PR so that they work regardless of the underlying implementation, merge them in, and then make sure this PR still passes with them in place?

With a BZ attached to this PR, I don't know how much pressure is on to take the time to do that though.